### PR TITLE
merging - project done

### PR DESCRIPTION
**Pull request by JonathanYK**
merging the finished project (for_pr) to an empty branch in order to create PR

### CODE REVIEW

File: .github/workflows/docker-image.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Docker Image CI
```
Comment (seniakalma) [inline .github/workflows/docker-image.yml:1]: CI is Continues Integration - Tests
This is part of the deployment (CD)

File: .github/workflows/docker-image.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+        
+    - name: Login to Docker Hub
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKER_HUB_USER }}
```
Comment (seniakalma) [inline .github/workflows/docker-image.yml:20]: Great secrets use!

File: README.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+
+>1. Generating a  session id (defined as random number with exact 9 digits), then adding it to db.
+>2. Receiving a session id and a json file, saving the relevant fields as a log (using both stream_handler and file_handlers - printing the log to a console/dedicated file) and saves the message of the session id to the db.
+
+Supported routes splitted to blueprints, 2 configuration files under `/src/instance` defined in order to execute this project both on test and production environments.
+Make sure to edit `/src/instance/key.txt` file with a random new key.
+In order to ensure that the access to db is thread safe - scoped_session of SQLAlchemy is used.
+
+The project has both unit and functionality tests, designed using Pytest framework.
```
Comment (seniakalma) [inline README.md:15]: Maybe worth adding them as a CI?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+## CD overview
+
+There is a docker-compose.yml with nginx support.  
+In order to build using the docker compose file, use the command:  
+`docker-compose up --build -d`  
+  
+The terraform code located in `/src/main.tf` creates aws environment with all required resources.
+
+On each push to the main branch, there is a github action that building and pushing the docker image to docker hub
```
Comment (seniakalma) [inline README.md]:
```suggestion
On each push to the main branch, there is a github action that builds and pushs the docker image to docker hub
```

File: README.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+Once on each day, there is crontab configuration (explained in `/reclone_project_and_docker_compose.sh`) that restarting the docker-compose:
```
Comment (seniakalma) [inline README.md]:
```suggestion
Once a day, there is crontab configuration (explained in `/reclone_project_and_docker_compose.sh`) that restarts the docker-compose:
```

File: reclone_project_and_docker_compose.sh [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+# add a scheduled task using crontab:
+# $crontab -e
+# Add  the following command (to be executed on each day at 00:15):
+# 15 0 * * * /home/ubuntu/reclone_shell/reclone_project_and_docker_compose.sh >> /home/ubuntu/cron.log 2>&1
+
+# or this one for each 5 minutes (for debug):
+# 0,5,10,15,20,25,30,35,40,45,50,55 * * * * /home/ubuntu/reclone_shell/reclone_project_and_docker_compose.sh >> /home/ubuntu/cron.log 2>&1
+
+# make sure to execute the following commands manually on the first time:
```
Comment (seniakalma) [inline reclone_project_and_docker_compose.sh:11]: Gr8

File: src/instance/flask.cfg [hunk 7]
```diff
@@ -0,0 +1,18 @@
+##########################################################
+#
+# This is a sample flask.cfg for developing the sessions Flask application
+#
+##########################################################
+import os
+
+# Read secret key from a local file
+f = open(os.getcwd() + '/instance/key.txt', 'r')
```
Comment (seniakalma) [inline src/instance/flask.cfg:9]: Great!

File: src/instance/key.txt [hunk 8]
```diff
@@ -0,0 +1 @@
+uqxhV2luxBARhhBqVAFKpCkM0Az8H0Y5
```
Comment (seniakalma) [inline src/instance/key.txt:1]: Remove from git repo (you can read/ask me why)

File: src/main.tf [hunk 9]
```diff
@@ -0,0 +1,145 @@
+terraform {
```
Comment (seniakalma) [inline src/main.tf:1]: Maybe seperated dir for tf files?

File: src/nginx/nginx.conf [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+events {
+    worker_connections  1024;
+    # worker_processes and worker_connections allows you to calculate maxclients value: 
+    # max_clients = worker_processes * worker_connections
+}
+
+http {
+    server {
+        listen 80;
+        #tells Nginx the hostname and the TCP port where it should listen for HTTP connections.
```
Comment (seniakalma) [inline src/nginx/nginx.conf]:
```suggestion
        # Tells Nginx the hostname and the TCP port where it should listen for HTTP connections.
```

File: src/tests/functionality/test_sessions.py [hunk 11]
```diff
@@ -0,0 +1,32 @@
+def test_main_page(test_client):
```
Comment (seniakalma) [inline src/tests/functionality/test_sessions.py:1]: You've done - can be really cool to add a short CI to run those tests - will strengthen GitHub actions

File: src/tests/unit/test_unit.py [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+    gen_session_id = test_send_json(test_client=test_client, json_msg_for_testing=json_msg_for_testing)
+
+    # get response message of gen_session_id:
+    response = test_client.get(f'/session_msg?session_id={gen_session_id}')
+
+    assert f"{json_msg_for_testing} session_id: " + str(gen_session_id) in str(response.data)
+
+
+# test session id without message:
+def test_sessions_id_without_msg(test_client):
```
Comment (seniakalma) [inline src/tests/unit/test_unit.py:130]: Great tests

File: src/timings.txt [hunk 13]
```diff
@@ -0,0 +1,53 @@
+Part A
```
Comment (seniakalma) [inline src/timings.txt:1]: Remember we talked about going first yourself on the PR?
It will always happen..

**Review by seniakalma — CHANGES_REQUESTED** [review]
Very good all in all
Really recommend to add a real CI

File: .github/workflows/docker-image.yml (hunk 2 above)
Comment (JonathanYK) [reply .github/workflows/docker-image.yml:20]: 👍

File: README.md (hunk 3 above)
Comment (JonathanYK) [reply README.md:15]: I'll add a CI

File: reclone_project_and_docker_compose.sh (hunk 6 above)
Comment (JonathanYK) [reply reclone_project_and_docker_compose.sh:11]: 👍

File: src/instance/flask.cfg (hunk 7 above)
Comment (JonathanYK) [reply src/instance/flask.cfg:9]: 👍

File: src/instance/key.txt (hunk 8 above)
Comment (JonathanYK) [reply src/instance/key.txt:1]: I'll read it and will try to understand - will talk about this on today's meeting.

File: src/main.tf (hunk 9 above)
Comment (JonathanYK) [reply src/main.tf:1]: will add

File: src/tests/functionality/test_sessions.py (hunk 11 above)
Comment (JonathanYK) [reply src/tests/functionality/test_sessions.py:1]: I'll add a CI

File: src/tests/unit/test_unit.py (hunk 12 above)
Comment (JonathanYK) [reply src/tests/unit/test_unit.py:130]: thank you!

File: src/timings.txt (hunk 13 above)
Comment (JonathanYK) [reply src/timings.txt:1]: I walk through it but missed this file :\

**Comment by JonathanYK** [comment]
All comments are integrated directly into the main branch. THANK YOU :)

**Pull request closed without merging**